R/translations.R

```diff
@@ -195,7 +202,10 @@ translations_direct_tidyverse <- list(
   dayofweek = str2lang("lubridate::wday"),
   hour = str2lang("lubridate::hour"),
   minute = str2lang("lubridate::minute"),
-  now = str2lang("lubridate::now")
+  now = str2lang("lubridate::now"),
+
+  # logic
+  choose = str2lang("dplyr::recode")
```

Comment:
I haven't used `dplyr::recode()` before, so I dug into this a bit. Looks good to me.

The only question here is whether we should set the argument `.default = NA` in the call to `recode()` to avoid a warning message in the case where the index is out of range. It would seem like yes, but `recode()` has the inconvenient behavior of enforcing type consistency, even for `NA`s. We can't know at parse time what type the output of `recode()` will have, so we can't know whether to use `NA`, `NA_real_`, `NA_character_`, `NA_integer_`, etc.  as the value of `.default`. So I think this should be kept as is.

Side note: Unfortunately for queryparser, `dplyr::case_when()` and `dplyr::coalesce()` also enforce type consistency like this.

Reply:
Ideally, we would use `.default = NA`. Since the default is `NA` anyway, perhaps we can just wrap with `suppressWarnings`? For example: `suppressWarnings(dplyr::recode(1:3, "a", "b"))`

Reply:
Let's leave it as is for now. In this case, I'm more concerned that `suppressWarnings()` might suppress a useful warning.